software/base_station_gui.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/local/bin/python3
+
+import sys
+from GUI import status_tab, controls_tab, map_tab
+from PyQt5.QtWidgets import *
+
+
+class BaseStationGUI(QMainWindow):
+    """
+    Main class for the Graphical aspect of the base station.
+
+    Creates the GUI window and then delegates the creation of the individual GUI tabs.
+
+    Created 13 September 2016
+    By Marcel van Workum
+
+    Modified 13 September 2016
+    By Marcel van Workum
+    """
+
+    APP_NAME = "TR - Base Station"
+    APP_WIDTH = 1280
+    APP_HEIGHT = 720
+    STATUS_BAR_BASE_MSG = "Rocket Status: "
+
+    def __init__(self):
+        """
+        Initialises the gui and tabs
+        """
+        super().__init__()
+
+        self.running = False
+
+        self.status_tab = QWidget()
+        self.controls_tab = QWidget()
+        self.map_tab = QWidget()
+
+        self.init_tabs()
+        self.init_gui()
+
+    def init_tabs(self):
+        """
+        Initialises the individual tabs and delegates their layout to the respective classes
+        """
+        tab_widget = QTabWidget(self)
+        tab_widget.setFixedSize(self.APP_WIDTH, self.APP_HEIGHT)
+        tab_widget.addTab(self.status_tab, "Rocket Status")
+        tab_widget.addTab(self.controls_tab, "Rocket Controls")
+        tab_widget.addTab(self.map_tab, "Rocket Location")
+        tab_widget.setCurrentIndex(2)
+
+        # Delegate creation of tabs to helper classes
+        status_tab.StatusTab()
+        controls_tab.ControlsTab()
+        map_tab.MapTab(self)
+
+    def init_gui(self):
+        """
+        Initialises the GUI window, and then shows it
+        """
+        self.statusBar().showMessage(self.STATUS_BAR_BASE_MSG + "Initialising")
+        self.setFixedSize(self.APP_WIDTH, self.APP_HEIGHT)
+        self.center_gui_window()
+        self.setWindowTitle(self.APP_NAME)
+        self.show()
+
+    def center_gui_window(self):
+        """
+        Helper method to center the QUI window in the screen
+        """
+        window_geometry = self.frameGeometry()
+        center_point = QDesktopWidget().availableGeometry().center()
+        window_geometry.moveCenter(center_point)
+        self.move(window_geometry.topLeft())
+
+    def closeEvent(self, event):
```

Comment:
Unfortunately not, it's overriding a super's method and for some reason they're camel cased. I think it's to mirror the actual cpp implementation of QT.